### ISSUE TITLE
Should writer.write() reject with the stored error when the writable stream is errored?

### ISSUE THREAD

**Issue by domenic** [whatwg/streams#700]
This makes sense to me, but apparently it currently rejects with a TypeError. See https://github.com/w3c/web-platform-tests/pull/5157#discussion_r106543376 .

**Comment by ricea** [comment]
I think it would be more consistent to reject with [[storedError]] and I know of no reason not to do so except that it is more code.

**Comment by ricea** [comment]
@tyoshino This is not urgent, but when you have time I would like your opinion on this.

**Comment by tyoshino** [comment]
Hmm ... I don't remember why I made it to do so.

In short, I think it's ok to fix it to reject with `[[storedError]]`.

I remember that somewhere I insisted that a stream should throw (reject with) an exception that indicates that the program is doing something just wrong than the stored exception when e.g. a stream is written to after requested to close.

A writable stream may be errored by either
- (1) `controller.error()` by the sink
- (2) `writer.abort()` by the user

The rationale applies to (2), but not for (1). I agree that we should use `[[storedError]]` for (1).

Regarding (2), I think we had some discussion and one option was to have a writable stream remember that it has been aborted. The idea behind my opinion was that it's weird if a user receives the error he/she passed to `writer.abort()` echo-ed back. IIRC, we chose not to have a dedicated state indicating that but have `WritableStreamFinishAbort()` set `[[storedError]]` to an exception indicating the stream has been aborted. So, `[[storedError]]` would hold an error telling the user that he/she's doing something wrong. So, using `[[storedError]]` is still fine for (2).
